### ISSUE TITLE
Crash on OS X El Capitan after recording audio

### ISSUE THREAD

**Issue by j3ffhubb** [j3ffhubb/musikernel#5]
100% Repeatable SEGFAULT on OS X simply by recording audio using the built-in microphones, doesn't happen on Linux or Windows which both work as expected.

Update:  It seems to be a crash in libsndfile, not related to recording at all.  It does not appear to be caused by python-wavefile, it may be something to do with llvm-clang or libsndfile itself.

**Comment by j3ffhubb** [comment]
I spent a considerable amount of time debugging this, the problem has been isolated, but I'm not sure how it could be fixed.  The problem occurs when sndfile.c:1712 tries to cast the received SNDFILE* to SF_PRIVATE* (via that hideous function macro).  The cast succeeds, but the any attempts to access members using -> give a SEGFAULT.  python-wavefile seems to be setting argstype and restype in an acceptable way, I tried the standard hacks to no avail.  I also tried editing the libsndfile source to skip the function macro and casting directly, but it still fails at the next access.

One possible fix for MusiKernel is to switch to a non-ctypes libsndfile wrapper like pysndfile, but I've not yet evaluated whether that works on all platforms, or how much effort the port would be.
